.changeset/stale-spoons-tie.md

```diff
@@ -0,0 +1,5 @@
+---
+"@remix-run/dev": patch
+---
+
+Avoid collisions between pathless layout routes and their nested index routes
```

Comment:
Could you add some of the context here that you included in the PR description? That'll help a ton when writing release notes.